### PR TITLE
refactored visit commands, added constants script

### PR DESCRIPTION
**Pull request by GeorgeSchafer**
# The Pull Request is ready

- [x] fixes #ISSUE_ID
- [X] all actions are passing
- [ ] only fixes a single issue

## Website

- [ ] mobile view is useable
- [ ] desktop view is useable
- [ ] no a-tags are used direktly (NavLink, MailLink, ExternalLink instead)
- [ ] all new texts are added to the translation files (at least the english one)
- [ ] tests have been added (if required)
- [ ] shared code has been extracted in a different file

I refactored some Cypress tests and consolidated re-used values as variables inside a ` constants.ts` file.

### CODE REVIEW

**Comment by GeorgeSchafer** [comment]
Further refactors, 2 spaces, rename of sponsors test

**Comment by GeorgeSchafer** [comment]
- [ ]  fixes #299

File: cypress/support/e2e.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+// ***********************************************************
```
Comment (Idrinth) [inline cypress/support/e2e.ts:1]: this file has no reason to be in the root

File: cypress/support/commands.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
+/// <reference types="cypress" />
```
Comment (Idrinth) [inline cypress/support/commands.ts:1]: this file has no reason to be in the root

File: cypress/fixtures/example.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (Idrinth) [inline cypress/fixtures/example.json:1]: this file has no reason to be in the root

File: cypress.config.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+export default {
```
Comment (Idrinth) [inline cypress.config.ts:1]: this file has no reason to be in the root

File: website/cypress/e2e/constants.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
+const PORT = '8080';
+const BASEURL = `http://localhost:${PORT}/`;
+const page = {
```
Comment (Idrinth) [inline website/cypress/e2e/constants.ts:3]: constants are meant to be in UPPER_SNAKE_CASE

File: website/cypress/e2e/constants.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  quickStart: 'quick-start/',
+  results: 'usage/results/',
+  support: 'support/',
+  routes: 'usage/routes/',
+  sponsors: 'contributing/sponsors/',
+  storage: 'usage/storage/',
+  usage: 'usage/'
+};
+
+export {
```
Comment (Idrinth) [inline website/cypress/e2e/constants.ts:21]: you can directly export the consts, no need for a separate object

**Review by Idrinth — CHANGES_REQUESTED** [review]

File: website/cypress/e2e/constants.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
+const PORT = '8080';
```
Comment (Idrinth) [inline website/cypress/e2e/constants.ts:1]: just noticed where this file is, please move it to fixtures or directly into the cypress folder

File: cypress.config.ts (hunk 4 above)
Comment (GeorgeSchafer) [reply cypress.config.ts:1]: Where is this located in the tree? I can't seem to find this. 

It looks like Cypress put these files there. Maybe they should be in .gitignore so as not to contaminate future builds?

File: cypress/support/e2e.ts (hunk 1 above)
Comment (GeorgeSchafer) [reply cypress/support/e2e.ts:1]: It looks like this file is necessary. Without it I get the following error
image

```
Your project does not contain a default supportFile. We expect a file matching cypress/support/e2e.{js,jsx,ts,tsx} to exist.

If a support file is not necessary for your project, set supportFile to false.

https://on.cypress.io/support-file-missing-or-invalid

[Stack trace](http://localhost:57920/__launchpad/index.html#)
Error: Your project does not contain a default supportFile. We expect a file matching cypress/support/e2e.{js,jsx,ts,tsx} to exist.

If a support file is not necessary for your project, set supportFile to false.

https://on.cypress.io/support-file-missing-or-invalid
    at L (:4165:9336)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async D.buildBaseFullConfig (:4388:64661)
    at async D.getFullInitialConfig (:4388:66110)
    at async D.setupNodeEvents (:4388:60237)
```

**Comment by GeorgeSchafer** [comment]
One last issue remaining, but it looks like it is in a file I had not touched.

**Comment by GeorgeSchafer** [comment]
`Run mv ./website/coverage ./
mv: cannot stat './website/coverage': No such file or directory
Error: Process completed with exit code 1.
`

File: .gitignore [hunk 8]
```diff
@@ -1,5 +1,9 @@
 # ROOT
 /node_modules
+cypress.config.ts
```
Comment (Idrinth) [inline .gitignore:3]: this will ignore any cypress.config.ts - not a good choice

File: .gitignore [hunk 9]
```diff
@@ -1,5 +1,9 @@
...
+
+# Cypress
+/cypress/support/commands.ts
```
Comment (Idrinth) [inline .gitignore:6]: this file must not exist, there is no reason to have cypress in the root

File: website/cypress/support/e2e.ts [hunk 10]
```diff
@@ -1 +0,0 @@
-import '@cypress/code-coverage/support';
```
Comment (Idrinth) [inline website/cypress/support/e2e.ts:1]: this is why the test coverage is no longer there

File: website/cypress/fixtures/constants.ts [hunk 11]
```diff
@@ -0,0 +1,19 @@
+export const BASE_URL = 'http://localhost:8080';
+export const page = {
```
Comment (Idrinth) [inline website/cypress/fixtures/constants.ts:2]: constants should be UPPER_SNAKE_CASE

File: cypress/support/e2e.ts (hunk 1 above)
Comment (Idrinth) [reply cypress/support/e2e.ts:1]: I'm not sure what you are doing there, likely trying to start cypress in the wrong folder. Cypress files in the root make no sense

**Review by Idrinth — CHANGES_REQUESTED** [review]

File: website/cypress/support/e2e.ts (hunk 10 above)
Comment (GeorgeSchafer) [reply website/cypress/support/e2e.ts:1]: I put it back.

File: cypress/support/e2e.ts (hunk 1 above)
Comment (GeorgeSchafer) [reply cypress/support/e2e.ts]: I see what happened. I removed this.

File: .gitignore (hunk 9 above)
Comment (GeorgeSchafer) [reply .gitignore:6]: Removed this root/cypress folder

**Comment by GeorgeSchafer** [comment]
I don't know where this change happened in package.json

```
    "spellchecker-cli": "^6.2.0",
    "markdownlint-cli": "^0.39.0",
    "@typescript-eslint/eslint-plugin": "^6.21.0",
```

I think I accidentally installed cypress to the root when the project opened in the root when I expected it to open inside /website. I went through and removed cypress, I will see if that helped.

**Comment by Idrinth** [comment]
> I think I accidentally installed cypress to the root when the project opened in the root when I expected it to open inside /website. I went through and removed cypress, I will see if that helped.

Most likely the issue at the root of all remaining things :) Other than those few left-overs it looks merge-ready now!

**Review by Idrinth — APPROVED** [review]

**Pull request merged**